hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroRecordMerger.java

```diff
@@ -59,6 +59,13 @@ private Option<IndexedRecord> combineAndGetUpdateValue(HoodieRecord older, Hoodi
       return Option.empty();
     }
 
+    if (newer.isDelete(schema, props)) {
+      if (newer.getOrderingValue(schema, props).compareTo(older.getOrderingValue(schema, props)) >= 0) {
+        return Option.empty();
```

Comment:
In some payload class like `EventTimeAvroPayload`, we already handle the deletion logic, but it lost the even time semantics somehow:

```java
    if (recordBytes.length == 0 || isDeletedRecord) {
      return Option.empty();
    }
```

We better still maintain the logic in the payload, not the merger.

Reply:
My rough idea is that if we are going to remove all kinds of payload classes, then we should start to move these logics bit by bit. I will document this idea somewhere.

Reply:
Meanwhile, this bug blocks some tests I am working on. So I will try to fix this for general Avro payload first.

Reply:
@yihua, @codope, do you guys have any comments?